src/lambda/expression.rs

```diff
@@ -75,13 +76,28 @@ impl Expression {
           match operation {
             Operation::Endpoint(req_template) => {
               let req = req_template.to_request(ctx)?;
-              let url = req.uri().clone();
+              let _url = req.uri().clone();
               let is_get = req.method() == reqwest::Method::GET;
+              let is_list = ctx
+                .context
+                .and_then(|c| c.ctx.path_node)
+                .and_then(|p| p.parent)
+                .map(|p| matches!(p.segment, QueryPathSegment::Index(_)))
+                .unwrap_or(false);
```

Comment:
`is_list` move to an external function or inside of `ctx.parent_is_list()`.  
Create an issue to write benchmarks and optimize this implementation.